pkg/scheduler/framework/v1alpha1/listers.go

```diff
@@ -38,11 +37,11 @@ type PodLister interface {
 // NodeInfoLister interface represents anything that can list/get NodeInfo objects from node name.
 type NodeInfoLister interface {
```

Comment:
These types should move to `framework/v1alpha1/interface.go`.
But the actual implementation could go to a separate subpackage (perhaps `framework/v1alpha1/listers`).
The rationale is that plugins and the cluster-autoscaler don't need the implementations.

But followup is fine.

Reply:
I don't like placing everything in one giant interface.go file. I agree we need to split the implementation and the interface into separate pkgs, and once in separate packages we should rename ```interface.go``` -> ```framework.go```, ```listers.go``` and ```types.go``` should stay (the WaitingPod, CycleState should move to ```types.go``` and remove their files)

Reply:
sgtm